nexus/inventory/src/builder.rs

```diff
@@ -351,11 +379,12 @@ impl CollectionBuilder {
 
     /// Record a collection error
     ///
-    /// This is used for operational errors encountered during the collection
-    /// process (e.g., a down MGS instance).  It's not intended for mis-uses of
-    /// this API, which are conveyed instead through returned errors (and should
-    /// probably cause the caller to stop collection altogether).
-    pub fn found_error(&mut self, error: anyhow::Error) {
+    /// See [`InventoryError`] for more on what kinds of errors are reported
+    /// this way.  These errors are stored as part of the collection so that
+    /// future readers can see what problems might make the collection
+    /// incomplete.  By contrast, [`CollectorBug`]s are not reported and stored
+    /// this way.
+    pub fn found_error(&mut self, error: InventoryError) {
```

Comment:
rustdoc is complaining about the doc links to private types; I'm surprised clippy isn't likewise complaining about `InventoryError`'s presence here in a `pub` API. Maybe make this (and the error types) `pub(crate)`?

Reply:
Thanks for looking.  I wound up exposing these instead.  At this point I think it's useful to be able to fully use a `CollectionBuilder` from the outside (for tests, if nothing else).